File: php/features.php
<?php
declare(strict_types=1);

// RandomUserGenerator SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RandomUserGeneratorFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RandomUserGeneratorBaseFeature();
            case "test":
                return new RandomUserGeneratorTestFeature();
            default:
                return new RandomUserGeneratorBaseFeature();
        }
    }
}

File: py/features.py
# RandomUserGenerator SDK feature factory

from feature.base_feature import RandomUserGeneratorBaseFeature
from feature.test_feature import RandomUserGeneratorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RandomUserGeneratorBaseFeature(),
        "test": lambda: RandomUserGeneratorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
